src/AdoPat/PatCache.cs

```diff
@@ -33,23 +33,15 @@ public PatCache(IStorageWrapper storage)
             this.cache = new Lazy<Dictionary<string, PatToken>>(() => this.ReadStorage());
         }
 
-        /// <summary>
-        /// Get a <see cref="PatToken"/> from the cache.
-        /// </summary>
-        /// <param name="key">The key for the target entry.</param>
-        /// <returns>The target value.</returns>
+        /// <inheritdoc/>
         public PatToken GetPat(string key)
         {
             PatToken token = null;
             this.cache.Value.TryGetValue(key, out token);
             return token;
         }
 
-        /// <summary>
-        /// Put a <see cref="PatToken"/> into the cache. May overwrite existing values.
-        /// </summary>
-        /// <param name="key">The key for this entry.</param>
-        /// <param name="patToken">The value for this entry.</param>
+        /// <inheritdoc/>
         public void PutPat(string key, PatToken patToken)

```

Comment:
I was re-thinking how this implementation would work in a concurrent execution scenario. If PutPat and ReadStorage were called simultaneously, this approach of writing and reading will result in mismatched data. Can this code ever be run in parallel? (concurrent tasks, threads, etc). If not, should we document that the implementation is not thread safe

Reply:
When all is said and done, no, this code will not ever run in parallel. Almost the entire application will be wrapped in a system-wide process lock. Doing any of these operations in parallel would certainly cause issues, but there should be little to no need to do that.

That said, if there's a standardized way to document that C# classes/methods are not threadsafe then I'm more than happy to do that.

Reply:
The XML docs support a Remarks section, which is where you will often see a note on thread [non]safety.

Reply:
I'll add a remark for the `PatClient`, the `PatCache`, and the `PatManager` saying that none of them is intended to be threadsafe.

Reply:
Oh, and no, @kyle-rader, the x-plat file lock is not automatically used in `StorageWrapper`, but my intention is to make the entire `PatManager.GetPatAsync` method wrapped in an x-plat lock from `Main`.